stubs/tensorflow/tensorflow/_internal.pyi

```diff
@@ -0,0 +1,5 @@
+# This includes internal types that are not part of the public API, but
+# are needed as annotations for the public API. Most commonly this is
+# internal base classes.
+
+class Trackable: ...
```

Comment:
I'm unsure on this choice. Trackable is a real class, but it's internal. Optimizer subclasses Trackable and there are other things that do subclass trackable and some public apis should have signature that looks like,

```python
def save_checkpoint(obj: Trackable) -> None
```

For a type like this is it preferred to keep it here or to add the real module it comes from even though that one is undocumented?

Reply:
I would prefer to put it in the real module that it lives in, if possible.

Reply:
Added. I did notice that module it lives in got changed between tf 2.8 -> 2.12 silently, not mentioned in release notes.